src/main/java/com/pingcap/tikv/TiSession.java

```diff
@@ -28,20 +29,23 @@
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
 import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
 
 
 public class TiSession implements AutoCloseable {
   private static final Map<String, ManagedChannel> connPool = new HashMap<>();
   private final TiConfiguration conf;
+  private final Function<CacheInvalidateEvent, Void> accumulatorFunction;
   // below object creation is either heavy or making connection (pd), pending for lazy loading
   private volatile RegionManager regionManager;
   private volatile PDClient client;
   private volatile Catalog catalog;
   private volatile ExecutorService indexScanThreadPool;
   private volatile ExecutorService tableScanThreadPool;
 
-  public TiSession(TiConfiguration conf) {
+  public TiSession(TiConfiguration conf, Function<CacheInvalidateEvent, Void> accumulatorFunction) {
     this.conf = conf;
+    this.accumulatorFunction = accumulatorFunction;
```

Comment:
Name is odd as well. What really here is an additional Error hook in general. Accumulator is just for TiSpark case. You can just name it as callback or listener.